Sources/MarkdownBabel/Execute/Evaluator/CodeToImageEvaluator.swift

```diff
@@ -0,0 +1,46 @@
+import Crypto
+
+import struct Foundation.Data
+import struct Foundation.URL
+
+public struct CodeToImageEvaluator: Evaluator, Sendable {
+	public let configuration: EvaluatorConfiguration
+	public let imageConfiguration: ImageEvaluationConfiguration
+	public let generateImageFileURL: GenerateImageFileURL
+
+	public init(configuration: EvaluatorConfiguration, generateImageFileURL: GenerateImageFileURL) {
+		precondition({ if case .codeBlock = configuration.executableMarkupType { true } else { false } }())
+		guard case .image(let imageConfiguration) = configuration.resultMarkupType else {
+			preconditionFailure("Tried to initialize image evaluator with non-image configuration: \(configuration)")
+		}
```

Comment:
I'd love to eventually avoid this and pull the information into the type, so that the "code to image evaluator" takes a code block and an image configuration.